vizro-core/src/vizro/models/_components/table.py

```diff
@@ -59,13 +60,47 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    # Interaction methods
+    def _get_figure_interaction_input(self) -> Dict[str, State]:
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "active_cell": State(component_id=self._callable_object_id, component_property="active_cell"),
+            "derived_viewport_data": State(
+                component_id=self._callable_object_id,
+                component_property="derived_viewport_data",
+            ),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
+
+    def _filter_interaction(
+        self, data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+    ) -> pd.DataFrame:
+        """Function to be carried out for pre-defined `filter_interaction`."""
+        ctd_active_cell = ctd_filter_interaction["active_cell"]
+        ctd_derived_viewport_data = ctd_filter_interaction["derived_viewport_data"]
+        if not ctd_active_cell["value"] or not ctd_derived_viewport_data["value"]:
+            return data_frame
+
+        # ctd_active_cell["id"] represents the underlying table id, so we need to fetch its parent Vizro Table actions.
+        source_table_actions = _get_component_actions(_get_parent_vizro_model(ctd_active_cell["id"]))
+
+        for action in source_table_actions:
+            if action.function._function.__name__ != "filter_interaction" or target not in action.function["targets"]:
+                continue
+            column = ctd_active_cell["value"]["column_id"]
+            derived_viewport_data_row = ctd_active_cell["value"]["row"]
+            clicked_data = ctd_derived_viewport_data["value"][derived_viewport_data_row][column]
+            data_frame = data_frame[data_frame[column].isin([clicked_data])]
+
+        return data_frame
+
     @_log_call
     def pre_build(self):
         if self.actions:
             kwargs = self.figure._arguments.copy()
 
             # This workaround is needed because the underlying table object requires a data_frame
-            kwargs["data_frame"] = DataFrame()
+            kwargs["data_frame"] = pd.DataFrame()
 
             # The underlying table object is pre-built, so we can fetch its ID.
             underlying_table_object = self.figure._function(**kwargs)
```

Comment:
Not related to this PR, but @petar-qb do you remember we had a conversation a while ago about trying to simplify all this stuff? I think we should try and do that still, especially now it's duplicated for `AGGrid` and `Table`.

Reply:
I really can't remember what we talked about exactly..
At least we can extract the implementation of this method and reuse it in the vm.AgGrid and vm.Table.

Also, I think we should get rid of "`if self.actions:`" statement in `pre_build` method and always conduct fetching of underlying id. In that way we can avoid raising a dash exception (something like: "Input of callback X doesn't exist."). This exception will be raised:
- if user provides an ID for the underlying table, 
- if the underlying table is an input (dash.State) of the custom action,
- if there is no "`actions`" defined for the vm.Table / vm.AgGrid

We should double check this assumption and if it's true we should also change one line in the `build` method. 
From:
```
                        dash_table.DataTable(**({"id": self._callable_object_id} if self.actions else {})), id=self.id
```
To:
```
                        dash_table.DataTable(id=self._callable_object_id), id=self.id
```

Reply:
My idea was that we can remove basically all of this, like so:
```
Index: vizro-core/src/vizro/models/_components/table.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vizro-core/src/vizro/models/_components/table.py b/vizro-core/src/vizro/models/_components/table.py
--- a/vizro-core/src/vizro/models/_components/table.py	(revision 4ea5b5f35cf9db4217537038d4f36ed4e8fc3ee4)
+++ b/vizro-core/src/vizro/models/_components/table.py	(date 1705597769502)
@@ -48,7 +48,9 @@
     # Convenience wrapper/syntactic sugar.
     def __call__(self, **kwargs):
         kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))
-        return self.figure(**kwargs)
+        figure = self.figure(**kwargs)
+        figure.id = self._callable_object_id
+        return figure
 
     # Convenience wrapper/syntactic sugar.
     def __getitem__(self, arg_name: str):
@@ -60,31 +62,14 @@
 
     @_log_call
     def pre_build(self):
-        if self.actions:
-            kwargs = self.figure._arguments.copy()
-
-            # This workaround is needed because the underlying table object requires a data_frame
-            kwargs["data_frame"] = DataFrame()
-
-            # The underlying table object is pre-built, so we can fetch its ID.
-            underlying_table_object = self.figure._function(**kwargs)
-
-            if not hasattr(underlying_table_object, "id"):
-                raise ValueError(
-                    "Underlying `Table` callable has no attribute 'id'. To enable actions triggered by the `Table`"
-                    " a valid 'id' has to be provided to the `Table` callable."
-                )
-
-            self._callable_object_id = underlying_table_object.id
+        self._callable_object_id = self.id + "***"
 
     def build(self):
         return dcc.Loading(
             html.Div(
                 [
                     html.H3(self.title, className="table-title") if self.title else None,
-                    html.Div(
-                        dash_table.DataTable(**({"id": self._callable_object_id} if self.actions else {})), id=self.id
-                    ),
+                    html.Div(dash_table.DataTable(id=self._callable_object_id), id=self.id),
                 ],
                 className="table-container",
                 id=f"{self.id}_outer",
```

Definitely not for this PR, but just wanted to make sure we didn't forget it.

Reply:
Ah yes, thanks for remembering 😄. This change really reduces the complexity of the table model ⚡.

Reply:
Ok will keep this un-resolved but will not implement in this PR